proto/src/lib.rs

```diff
@@ -107,9 +117,15 @@ pub mod serializers;
 /// // We expect a validation error here
 /// assert!(MyDomainType::decode(invalid_raw_bytes.as_ref()).is_err());
 /// ```
-pub trait Protobuf<T: Message + From<Self> + Default>
+pub trait Protobuf<T>
```

Comment:
I have modified the `Protobuf` trait such that it allows the message type to implement `TryFrom` instead of `From`. This allows safe parsing of the `TimeStamp` type, to prevent undefined behavior caused by integer overflow in DOS attack.

Reply:
Originally the idea here was that we wanted conversions from `SerializationType -> DomainType` to be fallible to allow for validation, and then from `DomainType -> SerializationType` to be infallible because `DomainType` should be a subset of `SerializationType`.

Changing this would imply that we could possibly have `DomainType`s that possibly do not have valid serialized representations, which implies to me that those `DomainType`s are incorrectly modeled, does it not?

So then the practical consequences of this architectural change would be:
1. All `From<DomainType> for SerializationType` impls now need to become `TryFrom<DomainType> for SerializationType`, touching all domain types
2. We need custom serde `Serialize` trait impls, as per https://github.com/informalsystems/tendermint-rs/pull/980/files#r717530415

This seems like a lot of work/code just to be able to cater for the `DomainTimestamp -> SerializedTimestamp` conversion, does it not? Is there no way to just implement a better `DomainTimestamp` whose conversion to `SerializedTimestamp` is infallible (as it should actually be)?